src/instructlab/utils.py

```diff
@@ -286,15 +292,23 @@ def _validate_documents(
             opened_files = False
             for pattern in file_patterns:
                 for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
-                    if os.path.isfile(file_path) and file_path.endswith(".md"):
-                        with open(file_path, "r", encoding="utf-8"):
-                            # Success! we could open the file.
-                            # We don't actually care about the contents, though.
-                            opened_files = True
+                    logger.debug("Checking file: %s", file_path)
+                    if os.path.isfile(file_path):
+                        if file_path.endswith(".md"):
+                            # Open markdown files in text mode
+                            with open(file_path, "r", encoding="utf-8"):
+                                # Successfully opened the markdown file.
+                                opened_files = True
+                        elif file_path.endswith(".pdf"):
```

Comment:
PDFs have a magic number, so we can determine if the file is really a pdf by checking the first 4 bytes of the header 
should we use that rather than rely on the file extension here?

Reply:
Personally I'm fine with the file extension, but no issue either way